core/src/main/java/org/opensearch/sql/ast/expression/NestedAllFields.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+package org.opensearch.sql.ast.expression;
+
+import java.util.Collections;
+import java.util.List;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.opensearch.sql.ast.AbstractNodeVisitor;
+import org.opensearch.sql.ast.Node;
+
+/**
+ * Represent the All fields which is been used in SELECT *.
+ */
+@ToString
+@EqualsAndHashCode(callSuper = false)
+public class NestedAllFields extends UnresolvedExpression {
+  @Getter
+  private final String path;
+
+  public NestedAllFields(String path) {
+    this.path = path.split("[.][*]")[0];
```

Comment:
```suggestion
    this.path = path.split("\\.")[0];
```

Reply:
Changed in https://github.com/Bit-Quill/opensearch-project-sql/pull/256/commits/14f410c441df9436a0d4605c063923224bb32be8